### DISCUSSION TITLE
Difference between planar and free-flyer joints

### DISCUSSION THREAD

**Discussion by LHT-Liu** [stack-of-tasks/pinocchio#2214] [Q&A]
Hello, i think i can use the planar joint for mobile manipulator, such as [this](https://github.com/stack-of-tasks/pinocchio/discussions/2184#discussioncomment-9064789), But what if my environment is not a plane, but a three-dimensional complex surface?

and i noticed that the `pin.JointModelFreeFlyer()` and `pin.JointModelComposite()` ，I want to know the difference between they and planar joint？

thanks

**Comment by stephane-caron** [comment]
> and i noticed that the pin.JointModelFreeFlyer() and pin.JointModelComposite() ，I want to know the difference between they and planar joint？

You should find the information you are looking for in [this page of the documentation](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/md_doc_c-maths_b-joints.html). If you have a more precise question after reading it, please summarize first the information you found for each joint, then feel free to ask.
